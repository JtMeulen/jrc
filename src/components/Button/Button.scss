@use '../../style/mixins.scss';

.button {
  @include mixins.font-family();

  font-weight: 700;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;

  &--primary {
    color: white;
    background-color: #1ea7fd;
  }

  &--secondary {
    color: #333;
    background-color: transparent;
    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
  }

  &--small {
    font-size: 0.8rem;
    padding: 10px 16px;
  }

  &--medium {
    font-size: 1rem;
    padding: 11px 20px;
  }

  &--large {
    font-size: 1.2rem;
    padding: 12px 24px;
  }
}
