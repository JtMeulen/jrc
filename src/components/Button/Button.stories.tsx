import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Button from "./Button";

export default {
  title: "Components/Button",
  component: Button,
  argTypes: {
    type: { control: 'inline-radio', options: ['primary', 'secondary']},
    size: { control: 'inline-radio', options: ['small', 'medium', 'large']},
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => (
  <Button
    {...args}
    onClick={(ev) => {
      console.log("onClick: ", ev.target);
    }}
  />
);

export const Primary = Template.bind({});
Primary.args = { label: "Primary", size: "large", type: "primary" };

export const Secondary = Template.bind({});
Secondary.args = { ...Primary.args, type: "secondary", label: "Secondary" };
