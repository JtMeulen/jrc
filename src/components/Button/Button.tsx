import React from "react";
import cn from "classnames";
import "./Button.scss";

export interface ButtonProps {
  type?: "primary" | "secondary" | "inverted";
  size?: "small" | "medium" | "large";
  label: string;
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const Button = ({
  type = "primary",
  size = "medium",
  label,
  onClick,
}: ButtonProps) => {
  return (
    <button
      className={cn("button", `button--${size}`, `button--${type}`)}
      onClick={onClick}
    >
      {label}
    </button>
  );
};

export default Button;
