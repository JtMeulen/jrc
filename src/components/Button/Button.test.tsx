import React from "react";
import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Button from "./Button";

describe("Button", () => {
  test("renders without error", () => {
    const { container } = render(<Button label="Test Button" />);
    expect(container).toMatchSnapshot();
  });

  test("renders with different props", async () => {
    const { container } = render(
      <Button label="Test Button" color="secondary" size="large" />
    );
    expect(container).toMatchSnapshot();
  });

  test("calls onClick handler when clicked", async () => {
    const handleClick = jest.fn();
    const { findByTestId } = render(
      <Button label="Test Button" onClick={handleClick} />
    );

    const buttonEl = await findByTestId("jrc-button");

    fireEvent.click(buttonEl);
    expect(handleClick).toHaveBeenCalled();
  });

  test("calls onClick handler with enter key", async () => {
    const handleClick = jest.fn();
    render(<Button label="Test Button" onClick={handleClick} />);

    await userEvent.keyboard("[Tab]");
    await userEvent.keyboard("[Enter]");
    expect(handleClick).toHaveBeenCalled();
  });
});
