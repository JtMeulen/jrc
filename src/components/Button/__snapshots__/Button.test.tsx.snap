// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with different props 1`] = `
<div>
  <button
    class="button button--large button--secondary"
    data-testid="jrc-button"
  >
    Test Button
  </button>
</div>
`;

exports[`Button renders without error 1`] = `
<div>
  <button
    class="button button--medium button--primary"
    data-testid="jrc-button"
  >
    Test Button
  </button>
</div>
`;
